src/trace/mod.rs

```diff
@@ -166,7 +166,8 @@ pub trait TraceReader {
         target.clear();
         target.insert(<Self::Time as timely::progress::Timestamp>::minimum());
         self.map_batches(|batch| {
-            target.clone_from(batch.upper());
+            target.clear();
+            target.extend(batch.upper().iter().cloned());
```

Comment:
This is a bit surprising as an improvement. I think `clone_from()` is supposed to do this already.

Reply:
I removed this change as it should be superseded by https://github.com/frankmcsherry/timely-dataflow/commit/7f5b8b75651d48380c3183195fb661383f660096

Reply:
For posterity: `clone_from()` does not do the expected thing; the derived implementation calls `clone()` rather than `clone_from()`.